<launch>
<arg name="Obstacle_NameSpace" default="ob3tacle_avoid" />
<arg name="USM_NameSpace" default="Up_Stair_Mode" />
<group ns="$(arg USM_NameSpace)">

    <rosparam param="stop_align_dis">120</rosparam>

    <rosparam param="far_dis">75</rosparam>

    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="close_dis">70</rosparam>
    <!-- threshold t3 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="err">10 </rosparam>


    <rosparam param="depth_imgshow">False</rosparam>
    <rosparam param="rgb_show">False</rosparam>
</group>

<group ns="$(arg Obstacle_NameSpace)">
  <!-- px1 is the leftmost point's x (175) -->
    <rosparam param="px1">150</rosparam>
    <!-- py1 is the leftmost point's y -->
    <rosparam param="py1">85</rosparam>
    <!-- px is columns of a block (100) -->
    <rosparam param="px">120</rosparam>
    <!-- py us rows of a block -->
    <rosparam param="py">50</rosparam>
    <!-- threshold t1 is the depth value,if it is bigger than t1 , then it is a obstacle point. -->
    <rosparam param="t1">0.01</rosparam>
    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="t2_c">160</rosparam>
    <!-- threshold t2 is the number of points ,if it is bigger than t2 , then it is a obstacle. -->
    <rosparam param="t2_o">1800</rosparam>

    <rosparam param="t2_oh">3900</rosparam>

    <!-- change the center velocity  -->
    <rosparam param="Vel_c">0.15</rosparam>

    <rosparam param="r1">0.5</rosparam>
    <rosparam param="r2">0.05</rosparam>
    <rosparam param="d2_min">0.2</rosparam>

    <rosparam param="laggy">10</rosparam>
    <rosparam param="moving_lock">True</rosparam>

    <!-- image_show turn on/off -->
    <rosparam param="depth_imgshow">True</rosparam>
    <rosparam param="rgb_show">True</rosparam>
</group>
 <node name="CrossFloorNavigation" pkg="image_fox" type="CrossFloorNavigation" output="screen">

    <rosparam param="xr_list">[-5.4400,-5.3400,-5.2400,-5.1400,-5.0400,-4.9400,-4.8400,-4.7400,-4.6400,-4.5400,-4.4400,-4.3400,-4.2400,-4.1400,-4.0400,-3.9400,-3.8400,-3.7400,-3.6400,-3.5400,-3.4400,-3.3400,-3.2400,-3.1400,-3.0400,-2.9400,-2.8400,-2.7400,-2.6400,-2.5400,-2.4400,-2.3400,-2.2400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400,-2.1400]</rosparam>
    <rosparam param="yr_list">[0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000]</rosparam>

    <rosparam param="xr_4f_turn">[0.0000,0.05,0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50,0.50]</rosparam>
    <rosparam param="yr_4f_turn">[0.0000,0.00,0.0,0.00,0.0,0.00,0.0,0.00,0.0,0.00,0.0,0.05,0.15,0.20,0.25,0.30,0.35,0.40,0.45,0.50,0.55,0.60,0.65,0.70,0.75,0.80,0.85,0.90,0.95,1.00,1.05,1.10,1.15,1.20,1.25,1.30,1.35,1.40,1.45,1.50,1.55,1.60,1.65,1.70,1.75]</rosparam>

    <rosparam param="xr_4f">[0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000,2.8000,2.9000,3.0000,3.1000,3.2000,3.3000,3.4000,3.5000,3.6000,3.7000,3.8000,3.9000,4.0000,4.1000,4.2000,4.3000,4.4000,4.5000,4.6000,4.7000,4.8000,4.9000,5.0000,5.1000,5.2000,5.3000,5.4000,5.5000,5.6000,5.7000,5.8000,5.9000,6.0000,6.1000,6.2000,6.3000,6.4000,6.5000,6.6000,6.7000,6.8000,6.9000,7.0000,7.1000,7.2000,7.3000,7.4000,7.5000,7.6000,7.7000,7.8000,7.9000,8.0000,8.1000,8.2000,8.3000,8.4000,8.5000,8.6000,8.7000,8.8000,8.9000,9.0000,9.1000,9.2000,9.3000,9.4000,9.5000,9.6000,9.7000,9.8000,9.9000,10.0000,10.1000,10.2000,10.3000,10.4000,10.5000,10.6000,10.7000,10.8000,10.9000,11.0000,11.1000,11.2000,11.3000,11.4000,11.5000,11.6000,11.7000,11.8000,11.9000,12.0000,12.1000,12.2000,12.3000,12.4000,12.5000,12.6000,12.7000,12.8000,12.9000,13.0000,13.1000,13.2000,13.3000,13.4000,13.5000,13.6000,13.7000,13.8000,13.9000,14.0000,14.1000,14.2000,14.3000,14.4000,14.5000,14.6000,14.7000,14.8000,14.9000,15.0000,15.1000,15.2000,15.3000,15.4000,15.5000,15.6000,15.7000,15.8000,15.9000,16.0000,16.1000,16.2000,16.3000,16.4000,16.5000,16.6000,16.7000,16.8000,16.9000,17.0000,17.1000,17.2000,17.3000,17.4000,17.5000,17.6000,17.7000,17.8000,17.9000,18.0000,18.1000,18.2000,18.3000,18.4000,18.5000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.6000,18.7000,18.8000,18.9000,19.0000,19.1000,19.2000,19.3000,19.4000,19.5000,19.6000,19.7000,19.8000,19.9000,20.0000,20.1000,20.2000,20.3000,20.4000,20.5000,20.6000,20.7000,20.8000,20.9000,21.0000,21.1000,21.2000,21.3000,21.4000,21.5000,21.6000]</rosparam>
    <rosparam param="yr_4f">[0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,-0.1000,-0.2000,-0.3000,-0.4000,-0.5000,-0.6000,-0.7000,-0.8000,-0.9000,-1.0000,-1.1000,-1.2000,-1.3000,-1.4000,-1.5000,-1.6000,-1.7000,-1.8000,-1.9000,-2.0000,-2.1000,-2.2000,-2.3000,-2.4000,-2.5000,-2.6000,-2.7000,-2.8000,-2.9000,-3.0000,-3.1000,-3.2000,-3.3000,-3.4000,-3.5000,-3.6000,-3.7000,-3.8000,-3.9000,-4.0000,-4.1000,-4.2000,-4.3000,-4.4000,-4.5000,-4.6000,-4.7000,-4.8000,-4.9000,-5.0000,-5.1000,-5.2000,-5.3000,-5.4000,-5.5000,-5.6000,-5.7000,-5.8000,-5.9000,-6.0000,-6.1000,-6.2000,-6.3000,-6.4000,-6.5000,-6.6000,-6.7000,-6.8000,-6.9000,-7.0000,-7.1000,-7.2000,-7.3000,-7.4000,-7.5000,-7.6000,-7.7000,-7.8000,-7.9000,-8.0000,-8.1000,-8.2000,-8.3000,-8.4000,-8.5000,-8.6000,-8.7000,-8.8000,-8.9000,-9.0000,-9.1000,-9.2000,-9.3000,-9.4000,-9.5000,-9.6000,-9.7000,-9.8000,-9.9000,-10.0000,-10.1000,-10.2000,-10.3000,-10.4000,-10.5000,-10.6000,-10.7000,-10.8000,-10.9000,-11.0000,-11.1000,-11.2000,-11.3000,-11.4000,-11.5000,-11.6000,-11.7000,-11.8000,-11.9000,-12.0000,-12.1000,-12.2000,-12.3000,-12.4000,-12.5000,-12.6000,-12.7000,-12.8000,-12.9000,-13.0000,-13.1000,-13.2000,-13.3000,-13.4000,-13.5000,-13.6000,-13.7000,-13.8000,-13.9000,-14.0000,-14.1000,-14.2000,-14.3000,-14.4000,-14.5000,-14.6000,-14.7000,-14.8000,-14.9000,-15.0000,-15.1000,-15.2000,-15.3000,-15.4000,-15.5000,-15.6000,-15.7000,-15.8000,-15.9000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000,-16.0000]</rosparam>

    <rosparam param="xr_4f2">[0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000,2.8000,2.9000,3.0000,3.1000,3.2000,3.3000,3.4000,3.5000,3.6000,3.7000,3.8000,3.9000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.0000,4.1000,4.2000,4.3000,4.4000,4.5000,4.6000,4.7000,4.8000,4.9000,5.0000,5.1000,5.2000,5.3000,5.4000,5.5000,5.6000,5.7000,5.8000,5.9000,6.0000,6.1000,6.2000,6.3000,6.4000,6.5000,6.6000,6.7000,6.8000,6.9000,7.0000,7.1000,7.2000,7.3000,7.4000,7.5000,7.6000,7.7000,7.8000,7.9000,8.0000,8.1000,8.2000,8.3000,8.4000,8.5000,8.6000,8.7000,8.8000,8.9000,9.0000,9.1000,9.2000,9.3000,9.4000,9.5000,9.6000,9.7000,9.8000,9.9000,10.0000,10.1000,10.2000,10.3000,10.4000,10.5000,10.6000,10.7000,10.8000,10.9000,11.0000,11.1000,11.2000,11.3000,11.4000,11.5000,11.6000,11.7000,11.8000,11.9000,12.0000,12.1000,12.2000,12.3000,12.4000,12.5000,12.6000,12.7000,12.8000,12.9000,13.0000,13.1000,13.2000,13.3000,13.4000,13.5000,13.6000,13.7000,13.8000,13.9000,14.0000,14.1000,14.2000,14.3000,14.4000,14.5000,14.6000,14.7000,14.8000,14.9000,15.0000,15.1000,15.2000,15.3000,15.4000,15.5000,15.6000,15.7000,15.8000,15.9000,16.0000,16.1000,16.2000,16.3000,16.4000,16.5000,16.6000,16.7000,16.8000,16.9000,17.0000,17.1000,17.2000,17.3000,17.4000,17.5000,17.6000,17.7000,17.8000,17.9000,18.0000,18.1000,18.2000,18.3000,18.4000,18.5000,18.6000,18.7000,18.8000,18.9000,19.0000,19.1000,19.2000,19.3000,19.4000,19.5000,19.6000,19.7000,19.8000,19.9000,20.0000,20.1000,20.2000,20.3000,20.4000,20.5000,20.6000,20.7000,20.8000,20.9000,21.0000,21.1000,21.2000,21.3000,21.4000,21.5000,21.6000,21.7000,21.8000,21.9000,22.0000,22.1000,22.2000,22.3000,22.4000,22.5000,22.6000,22.6000,22.6000,22.6000,22.6000,22.6000,22.6000,22.6000]</rosparam>
    <rosparam param="yr_4f2">[0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,-0.1000,-0.2000,-0.3000,-0.4000,-0.5000,-0.6000,-0.7000,-0.8000,-0.9000,-1.0000,-1.1000,-1.2000,-1.3000,-1.4000,-1.5000,-1.6000,-1.7000,-1.8000,-1.9000,-2.0000,-2.1000,-2.2000,-2.3000,-2.4000,-2.5000,-2.6000,-2.7000,-2.8000,-2.9000,-3.0000,-3.1000,-3.2000,-3.3000,-3.4000,-3.5000,-3.6000,-3.7000,-3.8000,-3.9000,-4.0000,-4.1000,-4.2000,-4.3000,-4.4000,-4.5000,-4.6000,-4.7000,-4.8000,-4.9000,-5.0000,-5.1000,-5.2000,-5.3000,-5.4000,-5.5000,-5.6000,-5.7000,-5.8000,-5.9000,-6.0000,-6.1000,-6.2000,-6.3000,-6.4000,-6.5000,-6.6000,-6.7000,-6.8000,-6.9000,-7.0000,-7.1000,-7.2000,-7.3000,-7.4000,-7.5000,-7.6000,-7.7000,-7.8000,-7.9000,-8.0000,-8.1000,-8.2000,-8.3000,-8.4000,-8.5000,-8.6000,-8.7000,-8.8000,-8.9000,-9.0000,-9.1000,-9.2000,-9.3000,-9.4000,-9.5000,-9.6000,-9.7000,-9.8000,-9.9000,-10.0000,-10.1000,-10.2000,-10.3000,-10.4000,-10.5000,-10.6000,-10.7000,-10.8000,-10.9000,-11.0000,-11.1000,-11.2000,-11.3000,-11.4000,-11.5000,-11.6000,-11.7000,-11.8000,-11.9000,-12.0000,-12.1000,-12.2000,-12.3000,-12.4000,-12.5000,-12.6000,-12.7000,-12.8000,-12.9000,-13.0000,-13.1000,-13.2000,-13.3000,-13.4000,-13.5000,-13.6000,-13.7000,-13.8000,-13.9000,-14.0000,-14.1000,-14.2000,-14.3000,-14.4000,-14.5000,-14.6000,-14.7000,-14.8000,-14.9000,-15.0000,-15.1000,-15.2000,-15.3000,-15.4000,-15.5000,-15.6000,-15.7000,-15.8000,-15.9000,-16.0000,-16.1000,-16.2000,-16.3000,-16.4000,-16.5000,-16.6000,-16.7000,-16.8000,-16.9000,-17.0000,-17.1000,-17.2000,-17.3000,-17.4000,-17.5000,-17.6000,-17.7000,-17.8000,-17.9000,-18.0000,-18.1000,-18.2000,-18.3000,-18.4000,-18.5000,-18.6000,-18.7000,-18.8000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.9000,-18.8000,-18.7000,-18.6000,-18.5000,-18.4000,-18.3000,-18.2000]</rosparam><rosparam param="xr_5f">[0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000,2.8000,2.9000,3.0000,3.1000,3.2000,3.3000,3.4000,3.5000,3.6000,3.7000,3.8000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000,3.9000]</rosparam>
    <rosparam param="xr_5f">[0.0000,0.1000,0.2000,0.3000,0.4000,0.5000,0.6000,0.7000,0.8000,0.9000,1.0000,1.1000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.3000,1.4000,1.5000,1.6000,1.7000,1.8000,1.9000,2.0000,2.1000,2.2000,2.3000,2.4000,2.5000,2.6000,2.7000,2.8000,2.9000,3.0000,3.1000,3.2000,3.3000,3.4000,3.5000,3.6000,3.7000,3.8000,3.9000,4.0000,4.1000,4.2000,4.3000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000]</rosparam>
    <rosparam param="yr_5f">[0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,0.0000,-0.1000,-0.2000,-0.3000,-0.4000,-0.5000,-0.6000,-0.7000,-0.8000,-0.9000,-1.0000,-1.1000,-1.2000,-1.3000,-1.4000,-1.5000,-1.6000,-1.7000,-1.8000,-1.9000,-2.0000,-2.1000,-2.2000,-2.3000,-2.4000,-2.5000,-2.6000,-2.7000,-2.8000,-2.9000,-3.0000,-3.1000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.3000,-3.4000,-3.5000,-3.6000,-3.7000,-3.8000,-3.9000,-4.0000,-4.1000,-4.2000,-4.3000,-4.4000,-4.5000,-4.6000,-4.7000,-4.8000,-4.9000,-5.0000,-5.1000,-5.2000,-5.3000,-5.4000,-5.5000,-5.6000,-5.7000,-5.8000,-5.9000,-6.0000,-6.1000,-6.2000,-6.3000,-6.4000,-6.5000,-6.6000,-6.7000,-6.8000,-6.9000,-7.0000,-7.1000,-7.2000,-7.3000,-7.4000,-7.5000,-7.6000,-7.7000,-7.8000,-7.9000,-8.0000,-8.1000,-8.2000,-8.3000,-8.4000,-8.5000,-8.6000,-8.7000,-8.8000,-8.9000,-9.0000,-9.1000,-9.2000,-9.3000,-9.4000,-9.5000,-9.6000,-9.7000,-9.8000,-9.9000,-10.0000,-10.1000,-10.2000,-10.3000,-10.4000,-10.5000,-10.6000,-10.7000,-10.8000,-10.9000,-11.0000,-11.1000,-11.2000,-11.3000]</rosparam>
    <rosparam param="xr_5f2">[4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.4000,4.3000,4.2000,4.1000,4.0000,3.9000,3.8000,3.7000,3.6000,3.5000,3.4000,3.3000,3.2000,3.1000,3.0000,2.9000,2.8000,2.7000,2.6000,2.5000,2.4000,2.3000,2.2000,2.1000,2.0000,1.9000,1.8000,1.7000,1.6000,1.5000,1.4000,1.3000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.2000,1.1000,1.0000,0.9000,0.8000,0.7000,0.6000,0.5000,0.4000,0.3000,0.2000,0.1000,0.0000]</rosparam>
    <rosparam param="yr_5f2">[-11.3000,-11.2000,-11.1000,-11.0000,-10.9000,-10.8000,-10.7000,-10.6000,-10.5000,-10.4000,-10.3000,-10.2000,-10.1000,-10.0000,-9.9000,-9.8000,-9.7000,-9.6000,-9.5000,-9.4000,-9.3000,-9.2000,-9.1000,-9.0000,-8.9000,-8.8000,-8.7000,-8.6000,-8.5000,-8.4000,-8.3000,-8.2000,-8.1000,-8.0000,-7.9000,-7.8000,-7.7000,-7.6000,-7.5000,-7.4000,-7.3000,-7.2000,-7.1000,-7.0000,-6.9000,-6.8000,-6.7000,-6.6000,-6.5000,-6.4000,-6.3000,-6.2000,-6.1000,-6.0000,-5.9000,-5.8000,-5.7000,-5.6000,-5.5000,-5.4000,-5.3000,-5.2000,-5.1000,-5.0000,-4.9000,-4.8000,-4.7000,-4.6000,-4.5000,-4.4000,-4.3000,-4.2000,-4.1000,-4.0000,-3.9000,-3.8000,-3.7000,-3.6000,-3.5000,-3.4000,-3.3000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.2000,-3.1000,-3.0000,-2.9000,-2.8000,-2.7000,-2.6000,-2.5000,-2.4000,-2.3000,-2.2000,-2.1000,-2.0000,-1.9000,-1.8000,-1.7000,-1.6000,-1.5000,-1.4000,-1.3000,-1.2000,-1.1000,-1.0000,-0.9000,-0.8000,-0.7000,-0.6000,-0.5000,-0.4000,-0.3000,-0.2000,-0.1000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000,-0.0000]</rosparam>

    <rosparam param="obstacle_avoid">false</rosparam>
    <!-- if amcl_modify=0,the testpath will not using vo  -->
    <rosparam param="amcl_modify">1</rosparam>

    <rosparam param="amcl_modify_epoch">3</rosparam>
    <!-- modify_range define the maxima range that visual odometry could correct. -->
    <rosparam param="modify_range">100</rosparam>
    <!-- turning time  -->
    <rosparam param="turning_time">8</rosparam>
    <!-- moving rate -->
    <rosparam param="moving_rate">2.5</rosparam>
</node>
<!-- launch nodes -->
 <!-- <include file ="$(find tracked_robot)/all_in_one.launch"></include> -->
<include file ="$(find tracked_robot)/odom_param.launch"></include>
 <node pkg="tracked_robot" name="Robot_Control_test" type="Robot_troller" />

  <!-- output="screen" -->

 <!-- <node pkg="rosaria" type="RosAria" name="RosAria" /> -->
 <!-- <node pkg="tf" type="static_transform_publisher" name="base_linktolaser" args="0 0 0 0 0 0 1 base_link laser 100" />
 <node pkg="hokuyo_node" type="hokuyo_node" name="hokuyo_node" output="screen"/> -->

  <!-- <node pkg="package-name" type="executable-name" name="node-name" /> -->
<!-- </node> -->

</launch>